x-pack/plugins/maps/public/connected_components/add_layer_panel/view.tsx

```diff
@@ -4,141 +4,193 @@
+    return this.state.layerSteps ? this.state.layerSteps[this.state.currentStepIndex] : null;
+  }
 
-    const newState: Partial<State> = {
-      isIndexingSource: false,
-    };
-    if (!keepSourceType) {
-      newState.layerWizard = null;
-      newState.importView = false;
-    }
-    // @ts-ignore
-    this.setState(newState);
+  _enableNextBtn = () => {
+    this.setState({ isNextStepBtnEnabled: true });
+  };
 
-    this.props.addPreviewLayers([]);
+  _disableNextBtn = () => {
+    this.setState({ isNextStepBtnEnabled: false });
   };
 
-  _onWizardSelect = (layerWizard: LayerWizard) => {
-    this.setState({ layerWizard, importView: !!layerWizard.isIndexingSource });
+  _startStepLoading = () => {
+    this.setState({ isStepLoading: true });
   };
 
-  _layerAddHandler = () => {
-    if (this.state.isIndexingSource && !this.props.isIndexingTriggered) {
-      this.props.setIndexingTriggered();
-    } else {
-      this.props.promotePreviewLayers();
-      if (this.state.importView) {
-        this.setState({
-          layerImportAddReady: false,
-        });
-        this.props.resetIndexing();
-      }
-    }
+  _stopStepLoading = () => {
+    this.setState({ isStepLoading: false });
   };
 
-  render() {
-    if (!this.props.flyoutVisible) {
+  _renderNextButton() {
+    const currentStep = this._getCurrentStep();
+    if (!currentStep) {
       return null;
     }
 
-    const panelDescription =
-      this.state.layerImportAddReady || !this.state.importView
-        ? i18n.translate('xpack.maps.addLayerPanel.addLayer', {
-            defaultMessage: 'Add layer',
-          })
-        : i18n.translate('xpack.maps.addLayerPanel.importFile', {
-            defaultMessage: 'Import file',
-          });
-    const isNextBtnEnabled = this.state.importView
-      ? this.props.isIndexingReady || this.props.isIndexingSuccess
-      : true;
+    let isDisabled = !this.state.isNextStepBtnEnabled;
+    let isLoading = this.state.isStepLoading;
+    if (currentStep.id === ADD_LAYER_STEP_ID) {
+      isDisabled = !this.props.hasPreviewLayers;
+      isLoading = this.props.isLoadingPreviewLayers;
+    } else {
+      isDisabled = !this.state.isNextStepBtnEnabled;
+      isLoading = this.state.isStepLoading;
+    }
 
     return (
-      <Fragment>
+      <EuiFlexItem grow={false}>
+        <EuiButton
+          data-test-subj="importFileButton"
+          disabled={isDisabled || isLoading}
+          isLoading={isLoading}
+          iconSide="right"
+          iconType={'sortRight'}
+          onClick={this._onNext}
+          fill
+        >
+          {currentStep.label}
+        </EuiButton>
+      </EuiFlexItem>
+    );
+  }
+
+  render() {
+    const currentStep = this._getCurrentStep();
+    return (
+      <>
         <EuiFlyoutHeader hasBorder className="mapLayerPanel__header">
           <EuiTitle size="s">
-            <h2>{panelDescription}</h2>
+            <h2>{currentStep ? currentStep.label : ADD_LAYER_STEP_LABEL}</h2>
```

Comment:
`currentStep` handling feels a little awkward since we're calling `_getCurrrentStep()` to obtain it as a form of derived state but then overriding it here saying that the actual current step is `ADD_LAYER_STEP_LABEL`. Seems like we should store the current step in the `state` and potentially update it via `componentDidUpdate()`.

Reply:
I have added currentStep to state. I also added a new const `SELECT_WIZARD_LABEL` to better reflect what this statement is doing. Its adding the page label when there is no current step.